### PR TITLE
Add order support and tests

### PR DESCRIPTION
**Pull request by bmw**
Built on #5374. Fixes a couple bugs and adds tests.

One thing I noticed here which I didn't fix is we don't extract the common name from the CSR. We should do this before the release, especially since users can provide their own CSR on the command line.

EDIT: This PR should be merged rather than squashed to preserve authorship.

### CODE REVIEW

File: acme/acme/client.py [hunk 1]
```diff
@@ -609,33 +620,42 @@ def poll_authorizations(self, orderr, deadline):
...
             response = self.net.get(orderr.uri)
             body = messages.Order.from_json(response.json())
             if body.error is not None:
-                raise IssuanceError(body.error)
+                raise errors.IssuanceError(body.error)
             if body.certificate is not None:
                 certificate_response = self.net.get(body.certificate).text
-                return orderr.update(fullchain_pem=certificate_response)
-        raise TimeoutError()
+                return orderr.update(body=body, fullchain_pem=certificate_response)
```
Comment (ohemorange) [inline acme/acme/client.py]: Finalize doesn't change anything in `body`?

Comment (bmw) [reply acme/acme/client.py]: Doesn't it? Earlier in the function, we post to the `finalize` endpoint and begin repeatedly making GET requests for the order until the `certificate` field is updated.

**Comment by bmw** [comment]
> As I mentioned OOB, this duplicates #5334 only you called it CommonClientTest and I called it ClientTestBase, oops.
>
> They're pretty much identical though so it shouldn't be a problem?

Yeah it'll be alright. There'll be some small merge conflicts, but it's nbd.

File: acme/acme/client.py (hunk 1 above)
Comment (ohemorange) [reply acme/acme/client.py]: Read this the other way around, thought you were removing body.

**Comment by ohemorange** [comment]
Cool, I'd like to double check the merge then, but otherwise looks good!

**Comment by bmw** [comment]
Oh you want to wait until we merge your PR and have me address merge conflicts here? That's fine. I can fix them.

**Comment by bmw** [comment]
Since there are multiple authors here, I'll want to fix up the commit history though so you won't have a clean commit to look at. Would you like me to push the non-rebased/squashed version somewhere for you to see?

**Comment by ohemorange** [comment]
I don't think it will be a problem, as long as the eventual PR is based on master.

**Comment by bmw** [comment]
PR rebased on master.

**Pull request merged**